src/stavetie.ts

```diff
@@ -6,9 +6,52 @@
 
 import { Vex } from './vex';
 import { Element } from './element';
+import { FontInfo } from './types/common';
+import { Note } from './note';
+import { Stave } from './stave';
+
+export interface StaveTieRenderOptions {
+  cp2: number;
+  last_x_shift: number;
+  tie_spacing: number;
+  cp1: number;
+  first_x_shift: number;
+  text_shift_x: number;
+  y_shift: number;
+  font: FontInfo;
+}
+
+export interface StaveTieRenderTieParams {
+  direction: number;
+  first_x_px: number;
+  last_x_px: number;
+  last_ys: number[];
+  first_ys: number[];
+}
 
 export class StaveTie extends Element {
-  constructor(notes, text) {
+  render_options: StaveTieRenderOptions;
+
+  protected text?: string;
+
+  protected font: FontInfo;
+  protected notes: {
```

Comment:
Lots of repetition here with this structure, can you make it a type?

Reply:
Done `Notes`